### ISSUE TITLE
Running this application on Linux

### ISSUE THREAD

**Issue by ghost** [akej74/grid-control#25]
Hi there,

So, I've been using this application without a hitch for a while now and it has served its purpose well. However, I'm now in the process of migrating from windows 7 to Linux Mint 18.3 and I'm wondering if there is any way to run this application on Linux. I know that it's possible to install the USB chip's driver (MCP2200) on Linux, but, afaik, the "Open Hardware Monitor" application doesn't work in that environment.

The best option I found is this: https://github.com/CapitalF/gridfan, but it provides a rather limited amount of control.

Any advice?

**Comment by akej74** [comment]
Hi,

The big problem running Grid Control on Linux is that Grid Control is depending on OpenHardwareMonitor, which as you said only runs on Windows. I'm pulling data from OHM with WMI (Windows Management Instrumentation) and everything is built around this setup. 

Porting to Linux would be a major rewrite of Grid Control and nothing I will have time to look into at the moment unfortunately. I believe "gridfan" is the best Linux implementation I'm aware of, but please also check [camctl](https://github.com/leaty/camctl).

One idea to check for interest for a Linux version could be if you asked on the NZXT subreddit, maybe there are some interest for making a fork of Grid Control and do a redesign for a Linux environment. Note that the user interface (QT) is fully compatible with Linux so that can be reused. It's mostly all things related to OHM that must be rewritten (as well as investigating how to monitor CPU/GPU temperatures in Linux etc).

**Comment by akej74** [comment]
One more project with Linux support, [leviathan](https://github.com/jaksi/leviathan). And one more, [hue-plus](https://github.com/kusti8/hue-plus).

**Comment by ghost** [comment]
Yep, that's what I thought. I'll keep investigating the matter. 

The best temp monitoring application for Linux that I could find is "GKrellM" (http://gkrellm.srcbox.net/). 
It was ported to Windows by Stefan Gehn with a few limitation. You can find the Windows version on his website (https://www.srcbox.net/projects/gkrellm/). This application can use either SpeedFan (http://www.almico.com/speedfan.php) or Core Temp (http://www.alcpu.com/CoreTemp/) to read the hardware sensors.

Interesting project. I'll try the NZXT subreddit like you suggested. 

Thanks!!!

**Comment by david-alleyn** [comment]
I am currently working on a port of Grid-Control for Linux. Thanks to Python/PyQT5, the amount of work required to make this work in Linux is pretty small.

The biggest hurdle is making Grid-Control more object oriented / modular. Currently, code which accesses sensor data and code which modifies the UI is scattered across modules. Providing access to multiple sensor platforms, without putting conditionals all over the application, requires it.

What works but hasn't been integrated yet:
- Reading temperature sensors from libsensors/lm-sensors and outputting them in a "tree-like" structure, similar to the WMI/OHM code.

**Comment by akej74** [comment]
Hi @david-alleyn , sounds good that you have made progress on the Linux version. I fully agree with you on the lack of modularity for Grid Control. This was one of my first projects learning Python and QT and many things could have been done differently, but you need to start somewhere... 

For some inspiration, please also have a look at [PyGrid by andyatgh](https://github.com/andyatgh/pygrid). This project was inspired by Grid Control but designed in a different way (more efficient, modular and OO as far as I can see having looked that the code briefly). There is also a [Reddit](https://www.reddit.com/r/NZXT/comments/6pf3w5/pygrid_opensource_fan_control_application_for/) discussion about his application.

Please let me know if you have any other questions about Grid Control for your Linux port.
